flink-runtime/src/main/scala/org/apache/flink/runtime/taskmanager/TaskManager.scala

```diff
@@ -129,6 +130,41 @@ class TaskManager(val connectionInfo: InstanceConnectionInfo,
     override def getValue: Double =
       ManagementFactory.getOperatingSystemMXBean().getSystemLoadAverage()
   })
+
+  // Preprocessing steps for registering cpuLoad
+  // fetch the method to get process CPU load
+  val getCPULoadMethod: Method = getMethodToFetchCPULoad()
+
+  // define the fetchCPULoad method as per the fetched getCPULoadMethod
+  val fetchCPULoad: (Any) => Double = if (getCPULoadMethod != null) {
+    (obj: Any) => getCPULoadMethod.invoke(obj).asInstanceOf[Double]
+  } else {
+    (obj: Any) => {
+      log.warning("getProcessCpuLoad method not available in the Operating System Bean" +
```

Comment:
Yes this was done due to one of the points mentioned by you in the earlier comments to log a full stack trace on each heartbeat (each time we collect metrics). I will remove it and make changes to log only once.